### PR TITLE
feat: expose express app for sub-app mounting

### PR DESCRIPTION
**Pull request by AndersDJohnson**
Exposes mockyeah's Express `app` instance to the external API so it can be mounted as a sub-app in a larger Express application (refer to "sub-app" documentation here: http://expressjs.com/en/api.html). This will enable people to use mockyeah transparently within their own servers, such as a development server that mocks some requests locally, while forwarding others on by actually proxying a backend with something like https://github.com/chimurai/http-proxy-middleware.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -36,5 +36,5 @@ module.exports = function Server(config, onStart) {
   };
 
   // Construct and return mockyeah API
-  return Object.assign({ server }, app.routeManager, { use, config, close });
+  return Object.assign({ server }, app, app.routeManager, { use, config, close });
```
Comment (AndersDJohnson) [inline server/index.js]: Wasn't sure if we prefer to merge this directly (risk of namespace conflicts), or make it `mockyeah.app`.

File: server/index.js [hunk 2]
```diff
@@ -30,11 +30,6 @@ module.exports = function Server(config, onStart) {
...
   });
 
-  // Expose ability to implement middleware via API
-  const use = function use() {
-    app.use.apply(app, arguments);
-  };
-
   // Construct and return mockyeah API
-  return Object.assign({ server }, app.routeManager, { use, config, close });
+  return Object.assign({ server }, app, app.routeManager, { config, close });
```
Comment (AndersDJohnson) [inline server/index.js]: Wasn't sure if we prefer to merge this directly (risk of namespace conflicts), or make it `mockyeah.app`.

We were already exposing `use` from `app`, so I can remove that explicit exposure now, if we go with this way of merging.

File: server/index.js (hunk 1 above)
Comment (ryanricard) [reply server/index.js]: I believe app instance conflicts with mock route methods (i.e. `.[get,post,put,delete,all]()`).

**Comment by ryanricard** [comment]
@AndersDJohnson exposing the express app for mounting on a parent application is interesting. I worry it couples mockyeah to Express. Thoughts?

File: server/index.js [hunk 3]
```diff
@@ -30,11 +30,6 @@ module.exports = function Server(config, onStart) {
     app.log(['serve', 'exit'], 'Goodbye.');
   });
 
-  // Expose ability to implement middleware via API
-  const use = function use() {
-    app.use.apply(app, arguments);
-  };
-
   // Construct and return mockyeah API
-  return Object.assign({ server }, app.routeManager, { use, config, close });
```
Comment (ryanricard) [inline server/index.js]: What is the reasoning to remove the `mockyeah.use()`? This could be a breaking change.

Comment (ryanricard) [reply server/index.js]: I've done some digging... I am not finding good reasoning for `.use()` being included in the mockyeah api. It is undocumented and seems to have creeped into the following commit: https://github.com/ryanricard/mockyeah/commit/fd161aa760a5b0b9099915cd7683cebd2204d006#diff-0861d6d6b50d7d695344bf2d86d6e5e6

**Comment by AndersDJohnson** [comment]
@ryanricard I don't think we can get more power without more tightly coupling to Express. I don't think we should limit the power of mockyeah due to these concerns. Express is the most popular server framework for Node.js. We could expose as `mockyeah.app` instead of just `mockyeah`, and then in future, if we change frameworks and want to support mounting as a sub-app in those frameworks, we can deprecate Express support and expose a different framework binding.

File: server/index.js (hunk 3 above)
Comment (AndersDJohnson) [reply server/index.js]: @ryanricard If we expose `app` via merging Express features into `mockyeah`, we don't need to specially expose `use`. If we go with `mockyeah.app` instead to expose, then we can leave `use` at the top level, unless you want to deprecate it at that level as well.

Comment (AndersDJohnson) [reply server/index.js]: @ryanricard It looks like there was an intent to expose the ability to add custom middleware - see [`test/integration/MiddlewareTest.js`](https://github.com/ryanricard/mockyeah/blob/master/test/integration/MiddlewareTest.js).

**Comment by AndersDJohnson** [comment]
@ryanricard Unless we want to add the proxying features, and any other future features, as features of mockyeah itself. This keeps the core lighter to mitigate against needing to support all use cases, while also not deterring the use of mockyeah when we don't support a feature.

**Pull request closed without merging**